components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -405,13 +335,125 @@ void AssetDiscoveryManager::MergeDiscoveredEthAssets(
     }
   }
 
-  CompleteDiscoverAssets(std::move(discovered_tokens),
-                         triggered_by_accounts_added);
+  std::move(callback).Run(std::move(discovered_tokens));
+}
+
+void AssetDiscoveryTask::DiscoverSPLTokensFromRegistry(
+    const std::vector<std::string>& account_addresses,
+    DiscoverAssetsCompletedCallback callback) {
+  // Convert each account address to SolanaAddress and check validity
+  std::vector<SolanaAddress> solana_addresses;
+  for (const auto& address : account_addresses) {
+    absl::optional<SolanaAddress> solana_address =
+        SolanaAddress::FromBase58(address);
+    if (!solana_address.has_value()) {
+      continue;
+    }
+
+    if ((*solana_address).IsValid()) {
+      solana_addresses.push_back(*solana_address);
+    }
+  }
+
+  if (solana_addresses.empty()) {
+    std::move(callback).Run({});
+    return;
+  }
+
+  const auto barrier_callback =
+      base::BarrierCallback<std::vector<SolanaAddress>>(
+          solana_addresses.size(),
+          base::BindOnce(&AssetDiscoveryTask::MergeDiscoveredSPLTokens,
+                         weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+  for (const auto& account_address : solana_addresses) {
+    json_rpc_service_->GetSolanaTokenAccountsByOwner(
```

Comment:
As mentioned in the DM, this API should have chain_id parameter.
Call this with mainnet directly here is fine at the moment, just make it clear that we only support mainnet so we just call it with mainnet here.